SU2_DOT/src/SU2_DOT.cpp

```diff
@@ -322,11 +325,15 @@ int main(int argc, char *argv[]) {
       /*--- If AD mode is enabled we can use it to compute the projection,
        *    otherwise we use finite differences. ---*/
 
-      if (config_container[iZone]->GetAD_Mode())
-        SetProjection_AD(geometry_container[iZone][INST_0], config_container[iZone], surface_movement[iZone] , Gradient);
-      else
+      if (config_container[iZone]->GetAD_Mode()) {
+        if (config_container[iZone]->GetSmoothGradient()) {
+          DerivativeTreatment_Gradient(geometry_container[iZone][INST_0], config_container[iZone], grid_movement[iZone], surface_movement[iZone] , Gradient);
+        } else {
+          SetProjection_AD(geometry_container[iZone][INST_0], config_container[iZone], surface_movement[iZone] , Gradient);
```

Comment:
And the DV smoothing happens instead of the DV projection, makes sense, but that method of the smoothing solver involves some deformation projections, so, does it also replace the normal projection to the surface?

Reply:
No. It does some deformation, but the standard projection to the surface has already happened at this point.